sources/cfsignal/src/service_check.rs

```diff
@@ -0,0 +1,64 @@
+use crate::error::{self, Result};
+use log::trace;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct ServiceHealth {
+    /// Whether or not the service is healthy.
+    pub(crate) is_healthy: bool,
+    /// In the event of an unhealthy service, the service's exit code (if found).
+    pub(crate) exit_code: Option<i32>,
+}
+
+pub(crate) trait ServiceCheck {
+    fn system_running(&self) -> Result<ServiceHealth>;
+}
+
+pub(crate) struct SystemdCheck {}
+
+impl ServiceCheck for SystemdCheck {
+    fn system_running(&self) -> Result<ServiceHealth> {
+        if is_system_running()? {
+            return Ok(ServiceHealth {
+                is_healthy: true,
+                exit_code: None,
+            });
+        }
+        Ok(ServiceHealth {
+            is_healthy: false,
+            exit_code: Some(Ok(1)?),
```

Comment:
(I am not a Rust expert) Does this need `Ok` or can it just be `Some(1)` since the type is `Option<i32>`?

Reply:
i am not rust expert too, as i wrote in main issue comment https://github.com/bottlerocket-os/bottlerocket/issues/1581#issuecomment-909325897, i am not rust programmer (this is first time that i put my hands on rust code)

Reply:
The `Ok(1)?` is synonymous with `1`, so it can be `exit_code: Some(1),`